src/packages/emmett-postgresql/src/eventStore/projections/pongo.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  type Event,
+  type EventTypeOf,
+  type ReadEvent,
+} from '@event-driven-io/emmett';
+import {
+  pongoClient,
+  type PongoClient,
+  type PongoDocument,
+} from '@event-driven-io/pongo';
+import pg from 'pg';
+import { type ProjectionDefintion } from './';
+
+export type PongoProjectionOptions<EventType extends Event> = {
+  documentId: (event: ReadEvent<EventType>) => string;
+  eventHandler: PongoProjectionHandler<EventType>;
+  eventTypes: EventTypeOf<EventType>[];
+};
+
+export type PongoProjectionHandler<EventType extends Event = Event> = (
+  documentId: (event: ReadEvent<EventType>) => string,
+  connectionString: string,
+  client: pg.PoolClient,
+  events: ReadEvent<EventType>[],
+) => Promise<void> | void;
```

Comment:
@oskardudycz Not caring about allocation, would it make sense to wrap all arguments in an object to make them position-independent?